### DISCUSSION TITLE
Handling right mouse clicks

### DISCUSSION THREAD

**Discussion by Rasanoob** [vkurko/calendar#234] [Q&A] [answered]
Is there a possibility to trigger the dateClick or eventClick functions with a right button click of the mouse? In my case I would like to have the possibility to add events when the left mouse button is clicked and if the right button is clicked I want to create a link from the week/day view to the corresponding event in the list view (scrolled to the correct date/time in the list view). I have tried to simulate a left click when a right click is happening but the functions don't trigger with this approach. Thanks in advance!

**Comment by vkurko** [comment, marked as answer]
Unfortunately [right click](https://github.com/vkurko/calendar/issues/195) is not yet implemented. But I think you might get some benefit from [dateFromPoint](https://github.com/vkurko/calendar?tab=readme-ov-file#datefrompoint-x-y-) function, which allows you to determine the time coordinates for any point inside the calendar inside your own click handler.

**Reply by Rasanoob** [reply]
Thanks for your quick answer. I think this can help indeed. As a last workaround I'm thinking of building a custom context menu when clicking left and let the user choose the next action.

**Reply by Rasanoob** [reply]
@vkurko your suggestion works very well, thanks! Now there is one strange thing I encounter here. The change of the view works fine and in the list view I have several events which I have to scroll. Now with element.scrollIntoView({ behavior: 'smooth', block: 'center' }); I try to scroll automatically to the event which the user clicked, but the scrolling doesn't work because the position of the elements turns out to be in the (screen) view. The following 2 statements: 
`rect = element.getBoundingClientRect();`
`scrollTop = window.pageYOffset || document.documentElement.scrollTop;`
are giving me for example 730 for rect and 0 for scrollTop although I have to scroll to see the actual element. Is it possible that the information about the height of the elements is not available at every time?

**Reply by vkurko** [reply]
Yes, both in Svelte itself and additionally in the calendar code, deferred tasks (e.g. microtasks) are widely used. Therefore, your code may not work if it is executed before the DOM is completely updated. I suggest you try scrolling in the [eventAllUpdated](https://github.com/vkurko/calendar?tab=readme-ov-file#eventallupdated) callback. Perhaps it will help you solve the problem.

**Reply by Rasanoob** [reply]
Thank you! eventAllUpdated didn't work but viewDidMount works fine.

**Reply by Rasanoob** [reply]
Hmm there is another strange behavior now. I have 2 different sort of events, the first sort is visible in the week view and the others should be visible in the list views. If I don't use viewDidMount they are loaded and shown correctly, but if I use viewDidMount the events that should be shown in the week view are also shown in the list view. Does the viewDidMount function somehow influence the event fetching? I solved this now with refetching events first, but I don't really understand why this is happening

**Reply by vkurko** [reply]
It's hard to tell what exactly is going on without seeing the code. If you can reproduce the problem somewhere in jsbin, codepen, etc. then I can help you.

**Reply by Rasanoob** [reply]
I have created a [demo](https://jsbin.com/gawetugake/edit?html,console,output) where I try to show the behavior. I'm loading 2 different sort of events in either the week view or the list view. Now when you are right clicking the day of the events you should see the list view of this day containing the other events. This works fine but if you look into the queryselector result you see that the elements of the old week view elements are in there although I've refetched the events first (the same thing is happening if I refetch in the eventhandler). As a result my scrolling is not correct because it is scrolling over the week view events instead of the list events (not doing this in the demo). 
As a side note: in the demo the eventAllUpdated callback is triggered perfectly fine while in my local code it is not working (at least I cannot see the print out in the console) I doen't really understand why this is.

**Reply by vkurko** [reply]
I tweaked your [demo](https://jsbin.com/socicaqosa/edit?html,output) and realized that `eventAllUpdated` is not ideal for such cases. I additionally had to use [loading](https://github.com/vkurko/calendar?tab=readme-ov-file#loading) in conjunction with `setTimeout` because `eventAllUpdated` is not called when there are no changes to the calendar after `refetchEvents` (for example, there were no events and no new events appeared). Some people may not like `setTimeout`, but it is an acceptable way to queue a macro task for execution.

**Reply by Rasanoob** [reply]
This works fine, thank you. I'm not really sure what you are doing here but it seems to work.
